### PR TITLE
Feat/client interface

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### Check the PR title.

- [ ] This PR title match the format: \<type\>(optional scope): \<description\>
- [ ] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -267,10 +238,10 @@ func (c *Client) execute(req *Request) (*Response, error) {
 	if hostHeader := req.Header.Get("Host"); hostHeader != "" {
```
Comment (ViolaPioggia) [inline client.go]: 这里也可以改成常量

File: easy_http.go [hunk 2]
```diff
@@ -17,147 +17,43 @@
...
+// For Example:
+//
+//	client, err := NewClient(config.WithDialer(customDialer))
+//	if err != nil {
+//		log.Fatalf("Failed to create client: %v", err)
+//	}
+//
+// Note: Uses standard library dialer by default for HTTPS support.
+func NewClient(opts ...config.ClientOption) (*Client, error) {
+	// 默认使用标准库以支持 https
```
Comment (ViolaPioggia) [inline easy_http.go]: 中文换成英文

File: easy_http.go [hunk 3]
```diff
@@ -17,147 +17,43 @@
...
+// MustNewClient creates a new client instance with given options.
+// It defaults to using the standard library for HTTPS support.
+//
+// For Example:
+//
+//	client := MustNewClient(config.WithTimeout(5 * time.Second))
+//
+// Note: It panics if client creation fails.
+func MustNewClient(opts ...config.ClientOption) *Client {
+	// 默认使用标准库以支持 https
```
Comment (ViolaPioggia) [inline easy_http.go]: 同上

File: client.go (hunk 1 above)
Comment (FGYFFFF) [reply client.go]: done, 等我一会批量刷一次，都改下

File: easy_http.go (hunk 3 above)
Comment (FGYFFFF) [reply easy_http.go]: done

File: easy_http.go (hunk 2 above)
Comment (FGYFFFF) [reply easy_http.go]: done

**Pull request merged**